internal/core/software_renderer/draw_functions.rs

```diff
@@ -881,9 +1423,9 @@ impl TargetPixel for crate::graphics::image::Rgb8Pixel {
 impl TargetPixel for PremultipliedRgbaColor {
     fn blend(&mut self, color: PremultipliedRgbaColor) {
         let a = (u8::MAX - color.alpha) as u16;
-        self.red = (self.red as u16 * a / 255) as u8 + color.red;
-        self.green = (self.green as u16 * a / 255) as u8 + color.green;
-        self.blue = (self.blue as u16 * a / 255) as u8 + color.blue;
+        self.red = ((self.red as u16 * a / 255) + color.red as u16).min(255) as u8;
+        self.green = ((self.green as u16 * a / 255) + color.green as u16).min(255) as u8;
+        self.blue = ((self.blue as u16 * a / 255) + color.blue as u16).min(255) as u8;
```

Comment:
I believe this might be a very performance sensitive operation.

Please run the printerdemo_mcu benchmark and see if this doesn't regress.

Maybe this could be improved using the Rust's builtin `saturating_add` function.

```sh
cargo bench -p printerdemo_mcu --
```